lib/ddtrace/configuration.rb

```diff
@@ -13,18 +18,50 @@ def configuration
     def configure(target = configuration, opts = {})
       if target.is_a?(Settings)
         yield(target) if block_given?
+
+        # Rebuild immutable components from settings
+        rebuild_components!(target)
+
+        target
       else
         PinSetup.new(target, opts).call
```

Comment:
Do we need to `rebuild_components!(target)` for the `PinSetup` case?

Reply:
I believe the answer is no.

When `Datadog.configure(object, ...)` is invoked, the target is not a settings object, and the `Pin` it configures is intended to be a one off configuration attached to the object in question. I think the use case for this is wholly independent than the global configuration case above it.